### PR TITLE
Annotation @phpstan-latte-template

### PR DESCRIPTION
**Pull request by MartinMystikJonas**

### CODE REVIEW

**Comment by MartinMystikJonas** [comment]
Just draft for now. I need to add docs and maybe some more tests

**Comment by MartinMystikJonas** [comment]
Ready for review

**Comment by lulco** [comment]
do we need all those empty files `{something}.php.latte`?

**Comment by MartinMystikJonas** [comment]
I am testing if the are resolved/checked based on annotation. I left them emptxy because absence of missing tempalte file... error is enough for my tests

File: docs/annotations.md [hunk 1]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
     }
 }
 ```
+
+## `@phpstan-latte-template`
+Allowed context: render call, method, class
+
+Value specifies what template or templates should be resolved and checked in given context.
+
+Multiple annotations can be used on single element or at different levels to check multiple tempaltes with same context.
```
Comment (lulco) [inline docs/annotations.md]: you have always typo in word templates :) but only in discussions / readme, in code it is usually correct

File: docs/annotations.md [hunk 2]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
...
+
+### Behaviour
+
+When this annotation is used on render call it will replace value passed as tempalte to render call.
```
Comment (lulco) [inline docs/annotations.md]: tempalte again

File: docs/annotations.md [hunk 3]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
...
+
+When annotation is used on method it will resolve given template in context of this method (alongside renders used in method body).
+
+When annotation is used on class it ats like it was added to all methods in class.
```
Comment (lulco) [inline docs/annotations.md]: acts?

File: docs/annotations.md [hunk 4]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
...
+
+### Annotation value
+
+Can use wildcard `*` to match multiple files.
```
Comment (lulco) [inline docs/annotations.md]: You can? or Wildcard can be used?

File: docs/annotations.md [hunk 5]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
...
+
+Can use placeholders:
+- `{dir}` = `__DIR__` (example `/app/dir/controls`)
+- `{file}` = `__FILE__` (example `/app/dir/controls/MyControl.php`)
+- `{fileName}` = `basename(__FILE__)` (example `MyControl.php`)
+- `{fileNameNoSuffix}` = file anme with suffix removed (example `MyControl`)
```
Comment (lulco) [inline docs/annotations.md]: we could naming like https://www.php.net/manual/en/function.pathinfo.php
filename, basename, extension

File: docs/annotations.md [hunk 6]
```diff
@@ -96,3 +97,89 @@ class MyControl extends Control
...
+- `{classShortName}` = class short name withnout namespace (example `MyControl`)
```
Comment (lulco) [inline docs/annotations.md]: I think we can call it className, and for FQN we can use fullClassName or so

**Review by lulco — APPROVED** [review]
lgtm

File: docs/annotations.md (hunk 1 above)
Comment (MartinMystikJonas) [reply docs/annotations.md]: Yes I have few such typos in muscle memory and cannot get rid of them

**Pull request merged**